livekit_agent.proto

```diff
@@ -74,38 +80,72 @@ enum JobStatus {
     JS_FAILED = 2;
 }
 
+message SimulateJobRequest {
+    JobType type = 1;
+    string room_id = 2;
+    string participant_id = 3;
+}
+
+message WorkerPing {
+    int64 timestamp = 1;
+}
+message WorkerPong {
+    int64 last_timestamp = 1;
+    int64 timestamp = 2;
+}
+
 message RegisterWorkerRequest {
     JobType type = 1;
     string worker_id = 2;
     string version = 3;
     string name = 4;
+    uint32 ping_interval = 5;
+    optional string namespace = 6;
 }
 
 message RegisterWorkerResponse {
     string worker_id = 1;
     string server_version = 2;
 }
 
+message MigrateJobRequest {
+    Job job = 1;
+}
+
 message AvailabilityRequest {
     Job job = 1;
+
+    // True when the job was previously assigned to another worker but has been 
+    // migrated due to different reasons (e.g. worker failure, job migration)
+    bool resuming = 2;
 }
 
 message AvailabilityResponse {
     string job_id = 1;
     bool available = 2;
+    bool should_retry = 3;
+
+    // How much the worker thinks the job will cost in terms of resources
+    double job_cost = 4;
 }
 
 message JobStatusUpdate {
     string job_id = 1;
     JobStatus status = 2;
     string error = 3;
     string user_data = 4;
+    
+    // The load that the job currently has on the worker
+    double load = 5;
+}
+
+message WorkerStatusUpdate {
+    WorkerStatus status = 1;
+    double load = 1;
+    int32 running_jobs = 2;
```

Comment:
I think sending ids instead of a count could allow us to reconcile the server understanding of jobs that are running on each client. otherwise if server keeps track of this separately.. they might get out of sync.

Reply:
Changed to a list of jobs but do you think this should be sent periodically?
WorkerStatusUpdate is a periodic update atm